frame/treasury/src/lib.rs

```diff
@@ -247,11 +302,19 @@ decl_storage! {
 				&<Module<T>>::account_id(),
 				T::RingCurrency::minimum_balance(),
 			);
+
+			// TODO: how to init both Ring and Kton in genesis?
+			//
+			// let _ = T::KtonCurrency::make_free_balance_be(
+			// 	&<Module<T>>::account_id(),
+			// 	T::KtonCurrency::minimum_balance(),
+			// );
```

Comment:
Methods like this doesn't have arguments to infer if we need to handle `RING` or `KTON`, this might need any flags?

Reply:
These are config, just 
```rust
let _ = T::KtonCurrency::make_free_balance_be(
 	&<Module<T>>::account_id(),
 	T::KtonCurrency::minimum_balance(),
);
```

Reply:
Get it, I'm trying to add some generic parameters for methods like `pot` now,

```rust
fn pot<T>() -> T::Balance where T: ... {

}
```

Reply:
Well, I think I probably understand how to master these naughty substrate traits now.

```rust
fn pot<C>() -> C::Balance 
where C: Currency<T::AccountId>,
{
    // ...
}
```

for tests: 

```rust
#[test]
fn gp() {
    let ring_balance = Treasury::pot::<Ring>::();
    let kton_balance = Treasury::pot::<Kton>::();
}
```

The usages of traits in substrate are quite like magic.